### PR TITLE
feat(react-components): Move AvatarGroup to stable

### LINKED ISSUE #24982: AvatarGroup: Move to stable
Move AvatarGroup to Stable once a partner validates it.

### PR DESCRIPTION
**Pull request by sopranopillow**
This PR moves AvatarGroup to stable and removes unstable warnings.

## Related Issue(s)

Fixes #24982

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |  `83.822 kB`<br />`21.029 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |    `48.692 kB`<br />`13.8 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |    `14.95 kB`<br />`5.989 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                                  |   `68.66 kB`<br />`19.138 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `188.681 kB`<br />`52.366 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `33.394 kB`<br />`11.007 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.851 kB`<br />`1.964 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/e44263f522e3e690c49ac69e5eef8e522e6e0a4a'>e44263f522e3e690c49ac69e5eef8e522e6e0a4a</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/Avatar-mount_base.html">1282</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/Avatar-mount.html">1265</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/Button-mount_base.html">931</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/Button-mount.html">921</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/FluentProvider-mount_base.html">1490</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/FluentProvider-mount.html">1493</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/FluentProviderWithTheme-mount_base.html">567</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/FluentProviderWithTheme-mount.html">569</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">542</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">545</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">579</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">572</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/MakeStyles-mount_base.html">1960</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/MakeStyles-mount.html">1940</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/SpinButton-mount_base.html">2296</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25005/perf-test-react-components/SpinButton-mount.html">2326</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by layershifter — COMMENTED** [review]
@sopranopillow @behowell I checked #22240 and I don't see that the component have been validated with partners yet. AFAIR we don't promote components to stable until they have been vetted in real apps. Can you please clarify?

**Comment by sopranopillow** [comment]
Reopening as a partner team has validated the component

File: packages/react-components/react-avatar/stories/Avatar/AvatarDefault.stories.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { ArgTypes } from '@storybook/api';
 import * as React from 'react';
-import { Avatar, AvatarProps } from '@fluentui/react-avatar';
+import { Avatar, AvatarProps } from '@fluentui/react-components';
```
Comment (khmakoto) [inline packages/react-components/react-avatar/stories/Avatar/AvatarDefault.stories.tsx:3]:
```suggestion
import { Avatar } from '@fluentui/react-components';
import type { AvatarProps } from '@fluentui/react-components';
```

**Review by khmakoto — APPROVED** [review]

**Review by micahgodbolt — APPROVED** [review]
🍻

**Pull request merged**